platform/ios/src/MGLMapView.mm

```diff
@@ -5375,33 +5374,50 @@ void onCameraDidChange(mbgl::MapObserver::CameraChangeMode mode) override {
     }
 
     void onWillStartLoadingMap() override {
-        [nativeView willStartLoadingMap];
+        [nativeView mapViewWillStartLoadingMap];
     }
 
     void onDidFinishLoadingMap() override {
-        [nativeView didFinishLoadingMap];
+        [nativeView mapViewDidFinishLoadingMap];
     }
 
-    void onDidFailLoadingMap() override {
-        [nativeView didFailLoadingMap];
+    void onDidFailLoadingMap(mbgl::MapObserver::ErrorType type, const std::string& message) override {
+        NSError *error;
+        NSString *errorDescription = @(message.c_str());
+        MGLErrorCode errorCode;
+        switch (type) {
+        case mbgl::MapObserver::ErrorType::ParseStyle:
+            errorCode = MGLErrorCodeParseStyleFailed;
+            break;
+        case mbgl::MapObserver::ErrorType::NotFound:
+            errorCode = MGLErrorCodeNotFound;
+            break;
+        case mbgl::MapObserver::ErrorType::LoadingStyle:
+            errorCode = MGLErrorCodeLoadStyleFailed;
```

Comment:
I'd prefer for map loading failure to have its own error type. However, we're no longer distinctively separating these in three different error types because `std::runtime_error` only provides a string.